pkg/depthmonitor/depthmonitor.go

```diff
@@ -128,60 +134,44 @@ func (s *Service) manage(warmupTime time.Duration) {
 		case <-time.After(manageWait):
 		}
 
-		// if we have crossed the adaptation, reset it
-		if time.Now().Sub(adaptationStart).Seconds() > adaptationWindow {
+		currentSize := float64(s.reserve.Size())
+		reserveNotHalfFull := isNotHalfFull(currentSize)
+
+		// if we have crossed 50% utilization, dont do anything
+		if !reserveNotHalfFull {
 			adaptationPeriod = false
+			continue
 		}
 
-		currentSize := float64(s.reserve.Size())
-		capacity := float64(s.reserve.Capacity())
-		reserveNotHalfFull := isNotHalfFull(currentSize, capacity)
-
 		// if we dont have 50% utilization of the reserve, enter into an adaptation period
 		// to see if we need to modify the depth to improve utilization
-		if reserveNotHalfFull && !adaptationPeriod {
+		if !adaptationPeriod {
 			adaptationPeriod = true
 			adaptationStart = time.Now()
 		}
 
-		// if we have crossed 50% utilization, dont do anything
-		if !reserveNotHalfFull {
-			adaptationPeriod = false
+		// if we have crossed the adaptation window, roll it back a little to allow sync to fill the reserve
+		if time.Since(adaptationStart).Seconds() > adaptationWindowSeconds {
+			adaptationStart = time.Now().Add(-time.Second * adaptationRollbackSeconds)
 		}
 
 		// based on the sync rate, determine the expected size of reserve at the end of the
 		// adaptation window
-		expectedSize := s.syncer.Rate()*(adaptationWindow-time.Now().Sub(adaptationStart).Seconds()) + currentSize
+		expectedSize := s.syncer.Rate()*(adaptationWindowSeconds-time.Since(adaptationStart).Seconds()) + currentSize
 
 		// if we are in the adaptation window and we are not expecting to have enough utilization
 		// by the end of it, we proactively decrease the storage depth to allow nodes to widen
 		// their neighbourhoods
-		if adaptationPeriod && isNotHalfFull(expectedSize, capacity) {
+		if isNotHalfFull(expectedSize) {
 			s.depthLock.Lock()
 			s.storageDepth--
-			s.topo.SetRadius(s.storageDepth)
-			s.logger.Infof("depthmonitor: reducing storage depth to better utilize reserve, new depth: %d", s.storageDepth)
+			s.topology.SetStorageDepth(s.storageDepth)
 			s.depthLock.Unlock()
-			adaptationPeriod = false
+			s.logger.Infof("depthmonitor: reducing storage depth to better utilize reserve, new depth: %d", s.storageDepth)
```

Comment:
The reason for this reset was that once we change the depth and allow the protocols to do their thing for `manageWait` time. Then, we should let the loop again determine if we need a new adaptation period. If we do, we should ideally calculate the new expectedSize based on the entire window and not just the remaining time in the window. This takes into account that the depth has already changed once.

If there is need to decrease further, I think we will anyway end up in another adaptation period and then check the expected size. If this is improved then no need to further change anything.

Otherwise, we would aggressively decrease the depth and then might need to increase it again causing some oscillation.